src/components/study-container.js

```diff
@@ -524,30 +472,128 @@ export function StudyContainer({ view, onChangeTab }) {
         // Note: studyUuid and dispatch don't change
     }, [studyUuid, dispatch, snackError, snackWarning]);
 
-    //handles map manual mode network reload
-    useEffect(() => {
-        if (mapManualRefresh && forceReloadNetwork) {
-            loadNetwork(true);
-        }
-    }, [forceReloadNetwork, loadNetwork, mapManualRefresh]);
-
     useEffect(() => {
         if (studyUuid) {
             loadTree();
         }
     }, [studyUuid, loadTree]);
 
+    function parseStudyNotification(studyUpdatedForce) {
+        const substationsIds =
+            studyUpdatedForce.eventData.headers['substationsIds'];
+        const substationsIdsArray = substationsIds
+            .substring(1, substationsIds.length - 1)
+            .split(', ');
+
+        const deletedEquipmentId =
+            studyUpdatedForce.eventData.headers['deletedEquipmentId'];
+        const deletedEquipmentType =
+            studyUpdatedForce.eventData.headers['deletedEquipmentType'];
+
+        return [
+            substationsIdsArray,
+            { id: deletedEquipmentId, type: deletedEquipmentType },
+        ];
+    }
+
+    useEffect(() => {
+        if (studyUpdatedForce.eventData.headers) {
+            if (
+                studyUpdatedForce.eventData.headers[UPDATE_TYPE_HEADER] ===
+                'study'
+            ) {
+                // study partial update :
+                // loading equipments involved in the study modification and updating the network
+                const [substationsIds, deletedEquipment] =
+                    parseStudyNotification(studyUpdatedForce);
+
+                // removing square brackets
+                if (substationsIds?.length > 0) {
+                    console.info('Reload network equipments');
+                    network.reloadImpactedSubstationsEquipments(substationsIds);
+                }
+
+                // removing deleted equipment from the network
+                if (deletedEquipment?.id && deletedEquipment?.type) {
+                    console.info(
+                        'removing equipment with id=',
+                        deletedEquipment?.id,
+                        ' and type=',
+                        deletedEquipment?.type,
+                        ' from the network'
+                    );
+                    network.removeEquipment(
+                        deletedEquipment?.type,
+                        deletedEquipment?.id
+                    );
+                }
+
+                dispatch(setUpdatedSubstationsIds(substationsIds));
+                dispatch(setDeletedEquipment(deletedEquipment));
```

Comment:
I would have separated this part of code in two useEffect: 
- one parsing study notification and dispatching substationsIds and deletedEquipment.
- another useEffect, triggered when substationsIds or deletedEquipment is modified in redux store to modify the network (reloadImpactedSubstationsEquipments and removeEquipment).

Reply:
Or maybe better, since mapEquipments and network are stored in redux store too, instead of modifying network here (and I think it's a bad practice to modify the state of the store without using dispatch) and mapEquipments in network-map-tab.js when substationsIds and deletedEquipment are dispatched, why not directly modify network and equipmentsMap in the reducer.js. This way you would not have to store deletedEquipment and substationsIds in the redux store.